### ISSUE TITLE
Webrtc crackling sound on multiple camera brands

### ISSUE THREAD

**Issue by borgqueenx** [AlexxIT/go2rtc#1164]
Whether i send audio mp3 files using the home assistant webrtc integration(and add the cameras in configuration.yaml as media players) or send tts voice messages, the sound is crackling like crazy. Sometimes the tts messages are hard to understand and sounds are very clunky and crackling.
A weird note to make is that this does not happen on the reolink duo 3, it plays sounds perfectly.
But on annke, hikvision and tapo cameras, the problem is the same. Sometimes(but rarely) sounds play normally without crackling up.
I already tried to add a delay to the sound of 2 seconds without luck.

I would love to find out with the community here how this is solvable.

**Comment by AlexxIT** [comment]
This can happen if you have chosen the wrong codec for your camera

**Comment by borgqueenx** [comment]
Which ones should i use for tapo and annke/hikvision and how do i select the codec? I assume annke and hikvision share the same codec. Maybe i need to transcode the file from mp3 to something else?

**Comment by AlexxIT** [comment]
You can find codec in stream info in go2rtc WebUI. You selecting codec while adding camera as media player.

**Comment by borgqueenx** [comment]
Under active producers it says the following at annke and hikvision cameras:
`Profile=1;\r\na=Media_header:MEDIAINFO=494D4B48010300000400050010710110401F000000FA000000000000000000000000000000000000;\r\na=appversion:1.0\r\n", "medias": [ "video, recvonly, H265", "audio, recvonly, PCMU/8000", "audio, sendonly, PCMU/8000, G726/8000, MPEG4-GENERIC/16000" ], "receivers": [ "96 H265, bytes=37299853942, senders=1", "0 PCMU/8000, bytes=325004160, senders=1" ], "senders": [ "0 PCMU/8000, bytes=18858, receivers=3" ], "recv": 37973941648, "send": 20362 },`

However when in configuration.yaml i copy MPEG4-GENERIC/16000 and G726/8000 as codec, i get a unknown error.

I used pcmu/8000 which worked, but is crackling.

**Comment by AlexxIT** [comment]
PCMU is right codec for this camera. Unfortunately, I don't have any Hikvision cameras to run a test.

**Comment by borgqueenx** [comment]
It is weird that also tapo cameras suffer from this, while using a other codec then hikvision and annke.
Is there a way to debug this?

**Comment by AlexxIT** [comment]
I have tapo. They used PCMA and works fine.

**Comment by borgqueenx** [comment]
But pretty much all my cameras are crackling. The only one that never did was my reolink duo 3. I have 3 annke cameras, 2 hikvision cameras and 9 tapo cameras (different models)

**Comment by NickM-27** [comment]
What is your go2rtc config

**Comment by borgqueenx** [comment]
> What is your go2rtc config

![Screenshot_20240606_225005_Home Assistant](https://github.com/AlexxIT/go2rtc/assets/20416925/54aaffdc-0923-4732-a64b-a7b8bfcf1f21)

A tapo and a annke camera. I dont have a pc to easily copy larger parts of the huge config file so hope this helps.
The annke camera has the mainstream(h265), substream(h264) and the tapo cams (only support h264) have the rtsp mainstream and the tapo protocol mainstream. Both cameras also have a ffmpeg aac encoding.

**Comment by NickM-27** [comment]
There is a known bug when using hikvision cameras that step backchannel will have crackling. You should use isapi instead

**Comment by borgqueenx** [comment]
> There is a known bug when using hikvision cameras that rtsp backchannel will have crackling. You should use isapi instead

Is this a bug with hikvision and annke or with webrtc?
How do i set up isapi with webrtc to send audio to cameras with home assistant?

**Comment by NickM-27** [comment]
See https://github.com/AlexxIT/go2rtc/issues/124

**Comment by NickM-27** [comment]
> for talking to people...i am using the webrtc integration to send audio files to the cameras.

It is used for both just like rtsp backchannel and that you what you need to use

Its what I've been using for a year

**Comment by borgqueenx** [comment]
> > for talking to people...i am using the webrtc integration to send audio files to the cameras.
> 
> It is used for both just like rtsp backchannel and that you what you need to use
> 
> Its what I've been using for a year

Does this integrate with webrtc and/or frigate?
So it works seamless.

**Comment by NickM-27** [comment]
It works exactly the same as before it's just a different way of getting audio to the camera

**Comment by borgqueenx** [comment]
> It works exactly the same as before it's just a different way of getting audio to the camera

Yes but how do we get this in there? The issue you linked has a python example script. But how do we use this for hikvision and annke?

**Comment by NickM-27** [comment]
https://github.com/AlexxIT/go2rtc#source-isapi

**Comment by borgqueenx** [comment]
Thanks! I just played around but when i press the mic button in frigate, a empty video icon shows 
![Screenshot_20240607_182730_Home Assistant](https://github.com/AlexxIT/go2rtc/assets/20416925/08e45a12-11d7-4108-9525-4139f576b42b)
![Screenshot_20240607_174140_Home Assistant](https://github.com/AlexxIT/go2rtc/assets/20416925/a8a10b65-4bb6-4856-87e4-da0b23e7861b)
